llama-index-core/llama_index/core/node_parser/interface.py

```diff
@@ -89,7 +90,13 @@ def _postprocess_parsed_nodes(
 
                 # update metadata
                 if self.include_metadata:
-                    node.metadata.update(parent_doc.metadata)
+                    # Update parent_doc.metadata with node.metadata, giving preference to node's values
+                    node.metadata = {**parent_doc.metadata, **node.metadata}
```

Comment:
How do we resolve conflicts if there are multiple child nodes all updating the same parent? It seems like it would be good to put it in a separate key and merge when you refer to the nodes. That way you get more control over whether you access only the parent attributes or not.

Reply:
Oh I think my comment applies to the chunk below. The comment in the code is incorrect- this line is merging parent metadata into the current node's metadata.